fix: failing due to RKE version retrieval logic

### Environment Information
<!--Please add information on the same line in quotes. Eg. - Terraform version: "v1.11.0" -->
- Terraform version: `v.1.12.2`
- Provider version: `v8.0.0-rc4`
- Rancher version: `head` https://github.com/rancher/rancher/commit/3e63ea6ff1f4a3c05176976c293d11f22b1eabcb
- Operating System of the Rancher Node: Ubuntu 24.04
- Operating System of the machine running Terraform: macOS
- Kubernetes Distro (k3s/rke2): RKE2
- Kubernetes Version: `v1.33.2+rke2r1`
- Infrastructure Provider (AWS/GCP/vSphere/etc): AWS
- What is the scope of the token given to the provider? Admin

### Describe the bug
When attempting to provision an RKE2/K3s cluster with `v8.0.0-rc4` in any recent `head` commit, the downstream cluster itself provisions. However, immediately after, you see the following:

```
rancher2_cluster_v2.cluster: Still creating... [6m10s elapsed]
rancher2_cluster_v2.cluster: Still creating... [6m20s elapsed]
rancher2_cluster_v2.cluster: Creation complete after 6m24s [id=fleet-default/mew-tf-rke2]
╷
│ Error: [ERROR] Listing RKE K8s System Images: Unknown schema type [rkeK8sSystemImage]
│ 
│   with rancher2_user.user,
│   on main.tf line 16, in resource "rancher2_user" "user":
│   16: resource "rancher2_user" "user" {

```
Additionally, if I try to run `terraform apply --auto-approve` again, I get this:

```
$ terraform apply --auto-approve
rancher2_cloud_credential.cloud_credential: Refreshing state... [id=cattle-global-data:cc-9h6pl]
rancher2_machine_config_v2.machine_config: Refreshing state... [id=fleet-default/nc-mew-tf-rke2-jqp94]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform planned the following actions, but then encountered a problem:

  # rancher2_global_role_binding.global_role_binding will be created
  + resource "rancher2_global_role_binding" "global_role_binding" {
      + annotations        = (known after apply)
      + global_role_id     = "user"
      + group_principal_id = (known after apply)
      + id                 = (known after apply)
      + labels             = (known after apply)
      + name               = "mew-tf-user"
      + user_id            = (known after apply)
    }

  # rancher2_user.user will be created
  + resource "rancher2_user" "user" {
      + annotations          = (known after apply)
      + enabled              = true
      + id                   = (known after apply)
      + labels               = (known after apply)
      + must_change_password = false
      + name                 = "mew-tf-user"
      + password             = (sensitive value)
      + principal_ids        = (known after apply)
      + username             = "mew-tf-user"
    }

Plan: 2 to add, 0 to change, 0 to destroy.
╷
│ Error: [ERROR] Listing RKE K8s System Images: Unknown schema type [rkeK8sSystemImage]
│ 
│   with rancher2_cloud_credential.cloud_credential,
│   on main.tf line 29, in resource "rancher2_cloud_credential" "cloud_credential":
│   29: resource "rancher2_cloud_credential" "cloud_credential" {
```

### To Reproduce
1. Setup Rancher on any `head` commit.
2. Run `terraform apply` against a main.tf (mine is below):

```
terraform {
  required_providers {
    rancher2 = {
      source = "terraform.local/local/rancher2"
      version = "8.0.0-rc4"
    }
  }
}

provider "rancher2" {
  api_url   = var.rancher_api_url
  token_key = var.rancher_admin_bearer_token
  insecure  = true
}

resource "rancher2_user" "user" {
  name     = var.username
  username = var.username
  password = var.password
  enabled  = true
}

resource "rancher2_global_role_binding" "global_role_binding" {
  name           = var.username
  global_role_id = var.global_role_id
  user_id        = rancher2_user.user.id
}

resource "rancher2_cloud_credential" "cloud_credential" {
  name = var.cloud_credential_name
  amazonec2_credential_config {
    access_key = var.aws_access_key
    secret_key = var.aws_secret_key
  }
}

resource "rancher2_machine_config_v2" "machine_config" {
  generate_name = var.machine_config_name
  amazonec2_config {
    ami            = var.ami
    region         = var.region
    security_group = [var.security_group]
    subnet_id      = var.subnet_id
    vpc_id         = var.vpc_id
    zone           = var.zone
  }
}

resource "rancher2_cluster_v2" "cluster" {
  name                                                       = var.cluster_name
  kubernetes_version                                         = var.kubernetes_version
  enable_network_policy                                      = false
  default_cluster_role_for_project_members                   = var.default_cluster_role_for_project_members
  default_pod_security_admission_configuration_template_name = var.default_pod_security_admission_configuration_template_name
  rke_config {
    machine_pools {
      name                         = var.machine_pool_etcd_name
      cloud_credential_secret_name = rancher2_cloud_credential.cloud_credential.id
      control_plane_role           = false
      etcd_role                    = true
      worker_role                  = false
      quantity                     = var.machine_pool_etcd_quantity
      machine_config {
        kind = rancher2_machine_config_v2.machine_config.kind
        name = rancher2_machine_config_v2.machine_config.name
      }
      rolling_update {
        max_surge       = 3
        max_unavailable = 3
      }
    }
    machine_pools {
      name                         = var.machine_pool_control_plane_name
      cloud_credential_secret_name = rancher2_cloud_credential.cloud_credential.id
      control_plane_role           = true
      etcd_role                    = false
      worker_role                  = false
      quantity                     = var.machine_pool_control_plane_quantity
      machine_config {
        kind = rancher2_machine_config_v2.machine_config.kind
        name = rancher2_machine_config_v2.machine_config.name
      }
      rolling_update {
        max_surge       = 3
        max_unavailable = 3
      }
    }
    machine_pools {
      name                         = var.machine_pool_worker_name
      cloud_credential_secret_name = rancher2_cloud_credential.cloud_credential.id
      control_plane_role           = false
      etcd_role                    = false
      worker_role                  = true
      quantity                     = var.machine_pool_worker_quantity
      machine_config {
        kind = rancher2_machine_config_v2.machine_config.kind
        name = rancher2_machine_config_v2.machine_config.name
      }
      rolling_update {
        max_surge       = 3
        max_unavailable = 3
      }
    }
  }
}
```

### Actual Result
Provisioning an RKE2/K3s cluster should work with `v8.0.0+` on any `head` commit.

### Expected Result
Provisioning an RKE2/K3s cluster does not work with `v8.0.0+` on any `head` commit.

--- a/rancher2/config.go
+++ b/rancher2/config.go
@@ -3,12 +3,10 @@ package rancher2
 import (
 	"context"
 	"fmt"
-	"sort"
 	"strings"
 	"sync"
 	"time"
 
-	"github.com/hashicorp/go-version"
 	"github.com/rancher/norman/clientbase"
 	"github.com/rancher/norman/types"
 	clusterClient "github.com/rancher/rancher/pkg/client/generated/cluster/v3"
@@ -148,39 +146,6 @@ func (c *Config) getK8SDefaultVersion() (string, error) {
 	}
 }
 
-func (c *Config) getK8SVersions() ([]string, error) {
-	if len(c.K8SSupportedVersions) > 0 {
-		return c.K8SSupportedVersions, nil
-	}
-
-	if c.Client.Management == nil {
-		_, err := c.ManagementClient()
-		if err != nil {
-			return nil, err
-		}
-	}
-
-	if ok, _ := c.IsRancherVersionLessThan(rancher2RKEK8sSystemImageVersion); ok {
-		return nil, nil
-	}
-
-	RKEK8sSystemImageCollection, err := c.Client.Management.RkeK8sSystemImage.ListAll(NewListOpts(nil))
-	if err != nil {
-		return nil, fmt.Errorf("[ERROR] Listing RKE K8s System Images: %s", err)
-	}
-	versions := make([]*version.Version, 0, len(RKEK8sSystemImageCollection.Data))
-	for _, RKEK8sSystem := range RKEK8sSystemImageCollection.Data {
-		v, _ := version.NewVersion(RKEK8sSystem.Name)
-		versions = append(versions, v)
-
-	}
-	sort.Sort(sort.Reverse(version.Collection(versions)))
-	for i := range versions {
-		c.K8SSupportedVersions = append(c.K8SSupportedVersions, "v"+versions[i].String())
-	}
-	return c.K8SSupportedVersions, nil
-}
-
 // Fix breaking API change https://github.com/rancher/rancher/pull/23718
 func (c *Config) fixNodeTemplateID(id string) string {
 	if ok, _ := c.IsRancherVersionGreaterThanOrEqual(rancher2NodeTemplateChangeVersion); ok && len(id) > 0 {
@@ -283,10 +248,6 @@ func (c *Config) ManagementClient() (*managementClient.Client, error) {
 	c.Client.Management = mClient
 
 	rancher2ClusterRKEK8SDefaultVersion, err = c.getK8SDefaultVersion()
-	if err != nil {
-		return nil, err
-	}
-	rancher2ClusterRKEK8SVersions, err = c.getK8SVersions()
 	if err != nil {
 		return nil, err
 	}

